test/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDaoMockTest.java

```diff
@@ -0,0 +1,213 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.sagebionetworks.bridge.TestConstants.TEST_STUDY;
+import static org.sagebionetworks.bridge.TestConstants.TEST_STUDY_IDENTIFIER;
+
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.TestUtils;
+import org.sagebionetworks.bridge.dao.CriteriaDao;
+import org.sagebionetworks.bridge.models.ClientInfo;
+import org.sagebionetworks.bridge.models.Criteria;
+import org.sagebionetworks.bridge.models.schedules.CriteriaScheduleStrategy;
+import org.sagebionetworks.bridge.models.schedules.Schedule;
+import org.sagebionetworks.bridge.models.schedules.ScheduleCriteria;
+import org.sagebionetworks.bridge.models.schedules.SchedulePlan;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.QueryResultPage;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+public class DynamoSchedulePlanDaoMockTest {
+    
+    private static final Set<String> ALL_OF_GROUPS = Sets.newHashSet("a","b");
+    private static final Set<String> NONE_OF_GROUPS = Sets.newHashSet("c","d");
+    
+    private DynamoSchedulePlanDao dao;
+    private DynamoDBMapper mapper;
+    private CriteriaDao criteriaDao;
+    private DynamoSchedulePlan mockPlan;
+    
+    @SuppressWarnings("unchecked")
+    @Before
+    public void before() {
+        mapper = mock(DynamoDBMapper.class);
+        criteriaDao = mock(CriteriaDao.class);
+        dao = new DynamoSchedulePlanDao();
+        dao.setSchedulePlanMapper(mapper);
+        dao.setCriteriaDao(criteriaDao);
+        
+        // We need the copy constructor to work in order to verify CriteriaDao works.
+        when(criteriaDao.copyCriteria(any(), any())).thenAnswer(invocation -> {
```

Comment:
Alternatively/additionally, move the copy constructor to the Criteria class, since it doesn't actually depend on implementation. (Implementation details are abstracted away behind Criteria.create().)